### ISSUE TITLE
MySQL and SQLite `upsert()`: `Internal error: Attempted to serialize empty result.`

### ISSUE THREAD

**Issue by christian-valadez** [prisma/prisma#19868]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v16.20.0           | 
| OS              | rhel-openssl-1.0.x |
| Prisma Client   | 4.14.1             |
| Query Engine    | d9a4c5988f480fa576d43970d5a23641aa77bc9c|
| Database        | mysql              |

Note: Using Planetscale 

## Logs
```
uest, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine requestBatch
prisma:client:libraryEngine requestBatch
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:request_handler {"clientVersion":"4.14.1"}
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:request_handler {"clientVersion":"4.14.1"}
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
```

## Client Snippet
```ts
      if (input.profilePhoto) {
        await prisma.cloudflareImage.upsert({
          where: {
            imageId: input.profilePhoto.imageId,
          },
          create: {
            imageId: input.profilePhoto.imageId,
            url: input.profilePhoto.url,
          },
          update: {
            imageId: input.profilePhoto.imageId,
            url: input.profilePhoto.url,
          },
        });
      }
```

## Schema
```prisma
model CloudflareImage {
  imageId String @id

  /// A Cloudflare Image URL in the form of: https://imagedelivery.net/<ACCOUNT_HASH>/<IMAGE_ID>/<VARIANT_NAME>
  url String @db.Text

  tilClass TilClass?

  post Post?
  User User?

  lastModified DateTime @updatedAt
  createdAt    DateTime @default(now())
}
```

## Prisma Engine Query
```
{"X":{}}
```

**Comment by janpio** [comment]
That code looks pretty innocent. Does this happen all the time? Can you reproduce this with the newest Prisma version as well?

**Comment by christian-valadez** [comment]
I can. I had run into it before w/ a previous version - https://github.com/prisma/prisma/issues/17279

So I hadn't upgraded in a while, but need to upgrade for interactive transactions functionality.

**Comment by luxaritas** [comment]
It looks like there's a handful of issues all related to this error on upsert: https://github.com/prisma/prisma/issues?q=is%3Aissue+is%3Aopen+Attempted+to+serialize+empty+result

It looks a little different from what I've seen reported so far, but I have a stable reproduction: https://github.com/luxaritas/prisma-serialize-empty-repro The upshot is:
1) Enable prisma relation mode
2) Create a model with two self-relations
3) Run an upsert twice (once to create, once to update)
4) Crash

**Comment by luxaritas** [comment]
The MySQL tag was added to this and other issues reporting this crash, though my repro shows this also crashes with sqlite fyi. It's not DB specific.

Given there continues to be new issues being filed about this crash, would it be useful to pick an issue and start marking others as duplicates to consolidate investigation?

**Comment by janpio** [comment]
Feel free to crosslink those if you are convinced they are caused by the same underlying reason.

**Comment by laplab** [comment]
Confirmed this on Prisma 5.10.2. Thank you @luxaritas for providing the [repro](https://github.com/luxaritas/prisma-serialize-empty-repro), that helped a lot! Running `index.ts` two times in a row indeed causes the engine to panic.

**Comment by janpio** [comment]
@christian-valadez Can you confirm that you were also using `relationMode=prisma` in your app?

**Comment by laplab** [comment]
This issue was fixed in [Prisma 5.13.0](https://github.com/prisma/prisma/releases/tag/5.13.0).
